### DISCUSSION TITLE
combine two loss

### DISCUSSION THREAD

**Discussion by liguopeng0923** [yoshitomo-matsubara/torchdistill#412] [Q&A] [answered]
Hi @yoshitomo-matsubara ，

I cannot find the file of combining two losses in your **legacy** folder. For example, what should I do when I want to train the models with cross entropy loss and KD loss in your new manner (\ie, **legacy** of Important Notice).

**Comment by yoshitomo-matsubara** [comment]
Hi @liguopeng0923 

Please use Q&A category for questions. (I changed the category here and the previous discussion)

There are many legacy config files that use a linear combination of multiple losses.
e.g. https://github.com/yoshitomo-matsubara/torchdistill/blob/main/configs/legacy/sample/ilsvrc2012/single_stage/ghnd/custom_resnet152_from_resnet152.yaml#L104-L149
https://github.com/yoshitomo-matsubara/torchdistill/blob/main/configs/legacy/sample/coco2017/multi_stage/ktaad/lraspp_mobilenet_v3_large_from_deeplabv3_resnet50.yaml#L228-L247

The KD loss that combines cross entropy and KL divergence is defined as `KDLoss` in torchdistill
e.g.,
https://github.com/yoshitomo-matsubara/torchdistill/blob/main/configs/legacy/sample/ilsvrc2012/single_stage/kd/alexnet_from_resnet152.yaml#L95-L99

Note: Please read [README](https://github.com/yoshitomo-matsubara/torchdistill/tree/main/configs) carefully for types of config files

**Comment by liguopeng0923** [comment]
I get it. Thank you!  But your examples are different types of losses (e.g. features). I want to combine multiple losses only according to outputs, not features.

**Reply by yoshitomo-matsubara** [reply, marked as answer]
Technically you can do that by extracting the last layer's output as a "feature"

**Reply by liguopeng0923** [reply]
ok，thanks!

**Reply by yoshitomo-matsubara** [reply]
When you close QA, please mark a reply as answer so that others can have better search experience if they have similar questions

**Comment by liguopeng0923** [comment]
And why we get a better result by appending a crossentropy loss in configs/sample/ilsvrc2012/kd/resnet18_from_resnet34.yaml

**Reply by yoshitomo-matsubara** [reply]
I don't understand the question. 

Also, please keep discussion in one thread as I requested before https://github.com/yoshitomo-matsubara/torchdistill/discussions/411#discussioncomment-7436826

**Reply by liguopeng0923** [reply]
Sorry for that.  I ignored that before.  I have no problem. Great work!
